packages/ui/src/forum/queries/forum.graphql

```diff
@@ -4,6 +4,27 @@ fragment ForumCategoryFields on ForumCategory {
   description
 }
 
+fragment ForumSubCategoryFields on ForumCategory {
+  id
+  title
+}
+
+fragment ForumCategoryDetailedFields on ForumCategory {
+  ...ForumCategoryFields
+  parent {
```

Comment:
So, [the runtime depth](https://github.com/Joystream/joystream/blob/40962ee609326d964b3985f68ed400754729c888/runtime/src/lib.rs#L619) is `6`. However, I wonder if we should do this query with every request for the category (to double-check with the wireframes and where we need breadcrumbs).

The question here is whether to query like that or try to take advantage of the Apollo query caching. Apollo client will cache the categories that were previously fetched IIRC it will cache based on the query/response fields.

So if we have a category tree `1 -> 2 -> 3 -> 4 -> 5` we could fetch one by one in the reverse order ( `5` then `4` then `3`, etc). This might be challenging using react hooks as we can't use hooks in loops.

A simple solution would be a multi-category query, that expands while getting more parents:

*   query categories `5` then
*   query categories `5,4` then
*   query categories `5,4,3` then
*   ... up to category without a parent, so it would stop with: `5,4,3,2,1`

However, this might re-query those for every category, as the starting query will always query the initial category (`5`). So, we'd need to investigate how [Apollo's cache works](https://www.apollographql.com/docs/react/caching/overview/). To see if my assumptions are correct or not.

Reply:
So for this PR I'd go with a simpler route:

*   create a dedicated `useForumBreadcrumbs()` hook
*   it will query for `GetCategoryBreadcrumbs` query (let's name it so no one would re-use it for other purposes)
*   The fields used in this query can be referenced as `ForumCategoryBreadcrumbFields` or not at all in this dedicated query:

```
query GetForumCategoryBreadcrumbs($id: ID!) {
  forumCategories(where:{id_eq: $id}) {
    id
    title
    parent {
      id
      title
      parent {
        id
        title
        parent {
          ... // up to 6th for now
        }
      }      
    }
  }
}
```

Reply:
This will create a proper abstraction over how we'll fetch this data. This hook will be used with/inside the `<ForumBreadcrumbs />` and to generate breadcrumbs at the top of the forum pages as well.

Reply:
I've created an issue for the performance review: #1158. I think that doing that after we have some real-world data might be better.